src/lib.rs

```diff
@@ -40,6 +42,7 @@ pub use crate::shims::intrinsics::EvalContextExt as IntrinsicsEvalContextExt;
 pub use crate::shims::os_str::EvalContextExt as OsStrEvalContextExt;
 pub use crate::shims::panic::{CatchUnwindData, EvalContextExt as PanicEvalContextExt};
 pub use crate::shims::sync::{EvalContextExt as SyncEvalContextExt};
+pub use crate::shims::threads::EvalContextExt as ThreadShimsEvalContextExt;
```

Comment:
We usually try to make our names consistent with rustc names, and the module is called `std::thread`, so could you call this `thread.rs`?